File: c/27.c
/* -59231 */
#include <stdio.h>
#include <stdlib.h>
#include "euler.h"

#define N 1000
#define M 80
#define V (M * M + M * N + N)

int main(int argc, char **argv)
{
    char *sieve;
    int a, b, c, n;
    int m = -1, ab = 0;
    prime_sieve(V, &sieve, NULL);
    for (a = 1 - N; a < N; a++)
    {
        for (b = 2; b < N; b++)
        {
            n = 0;
            c = b;
            while (c > 0 && sieve[c])
            {
                n++;
                c = n * n + a * n + b;
            }
            if (n > m)
            {
                m = n;
                ab = a * b;
            }
        }
    }
    printf("%d\n", ab);
    free(sieve);
    return 0;
}

File: py/27.py
# -59231
import euler
N = 1000
M = 80
m = -1
ab = 0
sieve = euler.prime_sieve(M * M + M * N + N)
for a in xrange(1-N, N):
    for b in xrange(2, N):
        n = 0
        while True:
            c = n * n + a * n + b
            if c < 2 or not sieve[c]:
                if n > m:
                    m = n
                    ab = a * b
                break
            n += 1
print ab
